src/temporal.jl

```diff
@@ -0,0 +1,23 @@
+"""
+    CustomTransform <: Transform
+
+Apply the function passed in to CustomTransform as the transform.
+"""
+struct HoD <: Transform end
+
+function _apply!(x, ::HoD; kwargs...)
+    x[:] = hour.(x)
```

Comment:
Does this even make sense as mutating? Tests currently error on the conversion from a `DateTime` array to now being one that would hold `Int`s

Reply:
Yeah, I don't think mutating make sense here.

Reply:
For that reason the set of methods for `HoD` could end up similar to `Periodic` transform with `TimeType` input.

Reply:
Aside: I made an issue about how to transform Tables for cases like this https://github.com/invenia/Transforms.jl/issues/12